pkg/services/ngalert/remote/remote_primary_forked_alertmanager.go

```diff
@@ -37,19 +37,26 @@ func (fam *RemotePrimaryForkedAlertmanager) GetStatus() apimodels.GettableStatus
 }
 
 func (fam *RemotePrimaryForkedAlertmanager) CreateSilence(ctx context.Context, silence *apimodels.PostableSilence) (string, error) {
-	return "", nil
+	_, err := fam.internal.CreateSilence(ctx, silence)
+	if err != nil {
+		return "", err
+	}
+	return fam.remote.CreateSilence(ctx, silence)
```

Comment:
As discussed offline, I think we can end up in situations where we have duplicate silences in the internal AM should the rpc fail to `fam.remote` fail and the call to `CreateSilence` retried.

I assume this is accounted for as while alerts are not being sent to the internal AM in this mode, when switching back there could be orphaned silences still in the internal AM that are not present in the remote AM.

Reply:
Looking at the design doc once more, I suppose since we send the complete state on startup and shutdown, the remote Alertmanager should eventually receive a full copy of any orphaned silences left in the Internal AM.

Reply:
Re-reading this I realize that it could be a problem. We send the complete state during **startup** (not shutdown) in Remote Primary mode.
Let's say we create a silence, it succeeds in the internal Alertmanager but not in the remote one.

```mermaid
sequenceDiagram
    actor User
    participant G as Grafana
    participant I as Internal AM
    participant R as Remote AM
    G ->> R: Send state (0 silences)
    User ->> G: Create Silence
    G ->> I: Create Silence
    I -->> G: OK
    G ->> R: Create Silence
    R -->> G: Error
    G -->> User: Error
    G -->> G: restart
    G ->> R: Send state (1 silence)
```


The silence was created in the internal Alertmanager, but the user received an error and expects no silences to be created. Next time Grafana restarts, the silence will be applied to the remote Alertmanager. The same goes for retries, we could end up restarting Grafana and having duplicate silences in the remote Alertmanager.